game-core/src/main/java/games/strategy/triplea/ai/flowfield/FlowFieldAI.java

```diff
@@ -0,0 +1,47 @@
+package games.strategy.triplea.ai.flowfield;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.framework.startup.ui.PlayerType;
+import games.strategy.triplea.ai.AbstractAi;
+import games.strategy.triplea.delegate.remote.IAbstractPlaceDelegate;
+import games.strategy.triplea.delegate.remote.IMoveDelegate;
+import games.strategy.triplea.delegate.remote.IPurchaseDelegate;
+import games.strategy.triplea.delegate.remote.ITechDelegate;
+
+public class FlowFieldAI extends AbstractAi {
```

Comment:
1. It seems like some documentation would be useful to know what/where this AI is going. At least something to start. My initial thought is a README file somewhere or a javadoc to propose an outline of what this AI is. What do you think of either of those two?
 
2. I wonder if we should "start this right" and make this the first AI that lives in an AI sub-project. That would give a hard, compile-time, boundary between the AI code and game-core.

Reply:
For 2, I think the only problem is `PlayerType.java`.  There isn't a way to dynamically add new player types.  They are currently hardcoded and need access to the Player class.  Do you have suggestions on how to make it dynamic?  I can work on that before this is merged and then create a new sub project.

Reply:
You could use the injections class to return the list of `PlayerType`.

For context, game-headed would depend on the AI project  and game-core, and the AI project would depend on just game-core. The game-headed project could grab the list of AI from the AI project and then set them in `Injections.java` which can then be used where needed. 

It's probably not that straight forward though as AI's are specifically referenced and used throughout the engine, so it is quite tightly coupled. We'd probably have to take a look to see how we can start decoupling. 

Ultimately making the `PlayerType` probably no longer an enum is needed. It would probably become just an interface, the game-headed package would instantiate the list of available AIs and inject them into `Injections.java`. In turn we'd have to look at the other locations of specific coupling, see if we can convert those to accept a list instead. For specific references to say 'WeakAI', that is something that we've wanted to dig out for some time.

If we can get at least some partial success here, that could be worth it while we continue to work to decouple AI from game-core. Meaning, even if it's just the one new AI stub in the AI sub-project, it is a step in the right direction.

Reply:
I'll check out the injections class.  I'm going to close this PR.